### ISSUE TITLE
GC crash in background_mark_simple1()

### ISSUE THREAD

**Issue by k15tfu** [dotnet/runtime#81154]
Hi!

.NET 7 app crashed as follows, unfortunately I cannot reproduce it so have only this:

```
Unhandled exception at 0x00007FFB87959B71 (coreclr.dll) in ....184504.dmp: 0xC0000005: Access violation reading location 0xFFFFFFFFFFFFFFFF.

>	coreclr.dll!SVR::gc_heap::background_mark_simple1(unsigned char * oo, int thread) Line 24942	C++
 	coreclr.dll!SVR::gc_heap::background_mark_simple(unsigned char * o, int thread) Line 25150	C++
 	coreclr.dll!SVR::gc_heap::background_promote(Object * * ppObject, ScanContext * sc, unsigned int flags) Line 25225	C++
 	[Inline Frame] coreclr.dll!ScanConsecutiveHandlesWithoutUserData(Object * * pValue, Object * *) Line 442	C++
 	coreclr.dll!BlockScanBlocksWithoutUserData(TableSegment * pSegment, unsigned int uBlock, unsigned int uCount, ScanCallbackInfo * pInfo) Line 554	C++
 	[Inline Frame] coreclr.dll!ProcessScanQNode(AsyncScanInfo *) Line 1230	C++
 	[Inline Frame] coreclr.dll!ProcessScanQueue(AsyncScanInfo * lParam, void(*)(AsyncScanInfo *, ScanQNode *, unsigned __int64)) Line 1200	C++
 	coreclr.dll!xxxTableScanQueuedBlocksAsync(HandleTable * pTable, TableSegment * pSegment, Holder<CrstBase *,&CrstBase::AcquireLock,&CrstBase::ReleaseLock,0,&CompareDefault<CrstBase *>,1> * pCrstHolder) Line 1311	C++
 	coreclr.dll!xxxAsyncSegmentIterator(HandleTable * pTable, TableSegment * pPrevSegment, Holder<CrstBase *,&CrstBase::AcquireLock,&CrstBase::ReleaseLock,0,&CompareDefault<CrstBase *>,1> * pCrstHolder) Line 1542	C++
 	coreclr.dll!TableScanHandles(HandleTable * pTable, const unsigned int * puType, unsigned int uTypeCount, TableSegment *(*)(HandleTable *, TableSegment *, Holder<CrstBase *,&CrstBase::AcquireLock,&CrstBase::ReleaseLock,0,&CompareDefault<CrstBase *>,1> *) pfnSegmentIterator, void(*)(TableSegment *, unsigned int, unsigned int, ScanCallbackInfo *) pfnBlockHandler, ScanCallbackInfo * pInfo, Holder<CrstBase *,&CrstBase::AcquireLock,&CrstBase::ReleaseLock,0,&CompareDefault<CrstBase *>,1> * pCrstHolder) Line 1698	C++
 	coreclr.dll!xxxTableScanHandlesAsync(HandleTable * pTable, const unsigned int * puType, unsigned int uTypeCount, TableSegment *(*)(HandleTable *, TableSegment *, Holder<CrstBase *,&CrstBase::AcquireLock,&CrstBase::ReleaseLock,0,&CompareDefault<CrstBase *>,1> *) pfnSegmentIterator, void(*)(TableSegment *, unsigned int, unsigned int, ScanCallbackInfo *) pfnBlockHandler, ScanCallbackInfo * pInfo, Holder<CrstBase *,&CrstBase::AcquireLock,&CrstBase::ReleaseLock,0,&CompareDefault<CrstBase *>,1> * pCrstHolder) Line 1793	C++
 	coreclr.dll!HndScanHandlesForGC(HandleTable * hTable, void(*)(Object * *, unsigned __int64 *, unsigned __int64, unsigned __int64) scanProc, unsigned __int64 param1, unsigned __int64 param2, const unsigned int * types, unsigned int typeCount, unsigned int condemned, unsigned int maxgen, unsigned int flags) Line 782	C++
 	[Inline Frame] coreclr.dll!Ref_TraceNormalRoots(unsigned int) Line 1059	C++
 	coreclr.dll!GCScan::GcScanHandles(void(*)(Object * *, ScanContext *, unsigned int) fn, int condemned, int sc, ScanContext * max_gen) Line 167	C++
 	coreclr.dll!SVR::gc_heap::background_mark_phase() Line 35016	C++
 	coreclr.dll!SVR::gc_heap::gc1() Line 21179	C++
 	coreclr.dll!SVR::gc_heap::bgc_thread_function() Line 36208	C++
 	[Inline Frame] coreclr.dll!`anonymous-namespace'::CreateSuspendableThread::__l2::<lambda_a030580c48363096550da8facec42d3a>::operator()(void *) Line 1414	C++
 	coreclr.dll!<lambda_a030580c48363096550da8facec42d3a>::<lambda_invoker_cdecl>(void * argument) Line 1419	C++
 	kernel32.dll!BaseThreadInitThunk()	Unknown
 	ntdll.dll!RtlUserThreadStart()	Unknown
```

It looks like the resulting `method_table(oo)` points to `0xFFFFFFFFFFFFFFFF`:
```
                if (overflow_p == FALSE)
                {
                    dprintf(3,("pushing mark for %Ix ", (size_t)oo));

                    go_through_object_cl (method_table(oo), oo, s, ppslot,
   00007FFB87959A1E  dec         dword ptr [rcx-7Dh]  
   00007FFB87959A21  in          eax,dx  
   00007FFB87959A22  adc         byte ptr [rbx+rdi-12h],cl  
   00007FFB87959A26  jae         SVR::gc_heap::background_mark_simple1+2B2h (07FFB879598B2h)  
   00007FFB87959A2C  jmp         $more_to_do+10h (07FFB87959FF1h)  
   00007FFB87959A31  mov         r11,qword ptr [rsi-10h]  
   00007FFB87959A35  add         r11,r15  
   00007FFB87959A38  add         r15,0FFFFFFFFFFFFFFF8h  
   00007FFB87959A3C  add         r15,rbp  
   00007FFB87959A3F  mov         qword ptr [start],r15  
   00007FFB87959A47  cmp         r11,r15  
   00007FFB87959A4A  jae         $more_to_do+10h (07FFB87959FF1h)  
   00007FFB87959A50  mov         r15,qword ptr [rsp+98h]  
   00007FFB87959A58  mov         rbp,r12  
   00007FFB87959A5B  neg         rbp  
   00007FFB87959A5E  mov         rsi,r13  
   00007FFB87959A61  nop         dword ptr [rax]  
   00007FFB87959A65  nop         word ptr [rax+rax]  
   00007FFB87959A70  mov         eax,dword ptr [rsi]  
   00007FFB87959A72  mov         r14d,dword ptr [rsi+4]  
   00007FFB87959A76  lea         rdi,[r11+rax*8]  
   00007FFB87959A7A  nop         word ptr [rax+rax]  
   00007FFB87959A80  mov         rax,qword ptr [r11]  
   00007FFB87959A83  prefetcht0  [rax]  
   00007FFB87959A86  cmp         rax,qword ptr [rbx+2228h]  
   00007FFB87959A8D  jb          SVR::gc_heap::background_mark_simple1+4DAh (07FFB87959ADAh)  
   00007FFB87959A8F  cmp         rax,qword ptr [rbx+2230h]  
   00007FFB87959A96  jae         SVR::gc_heap::background_mark_simple1+4DAh (07FFB87959ADAh)  
   00007FFB87959A98  mov         rcx,qword ptr [rbx+908h]  
   00007FFB87959A9F  mov         rdx,rax  
   00007FFB87959AA2  shr         rdx,9  
   00007FFB87959AA6  lea         r8,[rcx+rdx*4]  
   00007FFB87959AAA  mov         rcx,rax  
   00007FFB87959AAD  shr         rcx,4  
   00007FFB87959AB1  mov         edx,1  
   00007FFB87959AB6  and         cl,1Fh  
   00007FFB87959AB9  shl         edx,cl  
   00007FFB87959ABB  test        dword ptr [r8],edx  
   00007FFB87959ABE  jne         SVR::gc_heap::background_mark_simple1+4CBh (07FFB87959ACBh)  
   00007FFB87959AC0  lock or     dword ptr [r8],edx  
   00007FFB87959AC4  mov         ecx,1  
   00007FFB87959AC9  jmp         SVR::gc_heap::background_mark_simple1+4CDh (07FFB87959ACDh)  
   00007FFB87959ACB  xor         ecx,ecx  
   00007FFB87959ACD  test        ecx,ecx  
   00007FFB87959ACF  je          SVR::gc_heap::background_mark_simple1+5C1h (07FFB87959BC1h)  
   00007FFB87959AD5  jmp         SVR::gc_heap::background_mark_simple1+56Ah (07FFB87959B6Ah)  
   00007FFB87959ADA  test        rax,rax  
   00007FFB87959ADD  je          SVR::gc_heap::background_mark_simple1+5C1h (07FFB87959BC1h)  
   00007FFB87959AE3  cmp         rax,qword ptr [g_gc_lowest_address (07FFB87BE35D0h)]  
   00007FFB87959AEA  jb          SVR::gc_heap::background_mark_simple1+51Ah (07FFB87959B1Ah)  
   00007FFB87959AEC  cmp         rax,qword ptr [g_gc_highest_address (07FFB87BE35D8h)]  
   00007FFB87959AF3  jae         SVR::gc_heap::background_mark_simple1+51Ah (07FFB87959B1Ah)  
   00007FFB87959AF5  mov         rcx,qword ptr [SVR::gc_heap::min_segment_size_shr (07FFB87BE6460h)]  
   00007FFB87959AFC  mov         rdx,rax  
   00007FFB87959AFF  shr         rdx,cl  
   00007FFB87959B02  mov         rcx,qword ptr [SVR::seg_mapping_table (07FFB87BE6BE8h)]  
   00007FFB87959B09  imul        rdx,rdx,0B0h  
   00007FFB87959B10  mov         r8,qword ptr [rdx+rcx+40h]  
   00007FFB87959B15  test        r8,r8  
   00007FFB87959B18  jne         SVR::gc_heap::background_mark_simple1+524h (07FFB87959B24h)  
   00007FFB87959B1A  mov         rcx,qword ptr [SVR::gc_heap::g_heaps (07FFB87BE6BB0h)]  
   00007FFB87959B21  mov         r8,qword ptr [rcx]  
   00007FFB87959B24  cmp         rax,qword ptr [r8+2228h]  
   00007FFB87959B2B  jb          SVR::gc_heap::background_mark_simple1+5C1h (07FFB87959BC1h)  
   00007FFB87959B31  cmp         rax,qword ptr [r8+2230h]  
   00007FFB87959B38  jae         SVR::gc_heap::background_mark_simple1+5C1h (07FFB87959BC1h)  
   00007FFB87959B3E  mov         rcx,qword ptr [rbx+908h]  
   00007FFB87959B45  mov         rdx,rax  
   00007FFB87959B48  shr         rdx,9  
   00007FFB87959B4C  lea         r8,[rcx+rdx*4]  
   00007FFB87959B50  mov         rcx,rax  
   00007FFB87959B53  shr         rcx,4  
   00007FFB87959B57  mov         edx,1  
   00007FFB87959B5C  and         cl,1Fh  
   00007FFB87959B5F  shl         edx,cl  
   00007FFB87959B61  test        dword ptr [r8],edx  
   00007FFB87959B64  jne         SVR::gc_heap::background_mark_simple1+5C1h (07FFB87959BC1h)  
   00007FFB87959B66  lock or     dword ptr [r8],edx  
   00007FFB87959B6A  mov         r10,qword ptr [rax]  
   00007FFB87959B6D  and         r10,0FFFFFFFFFFFFFFF8h  
-> 00007FFB87959B71  mov         ecx,dword ptr [r10]  
   00007FFB87959B74  test        ecx,ecx  
   00007FFB87959B76  jns         SVR::gc_heap::background_mark_simple1+585h (07FFB87959B85h)  
   00007FFB87959B78  mov         r9d,dword ptr [rax+8]  
   00007FFB87959B7C  movzx       ecx,cx  
[...]
```

.NET 7.0.0
Windows x64

**Comment by mangod9** [comment]
Hi @k15tfu, this looks similar to https://github.com/dotnet/runtime/issues/78206 which has been fixed in 7.0.2. Could you please try upgrading to check if that fixes it for you?

**Comment by mangod9** [comment]
Going to close this assuming this a dupe. If it repros in 7.0.2 please reopen
